### DISCUSSION TITLE
ERROR: Error executing 'Rscript'

### DISCUSSION THREAD

**Discussion by royfrancis** [quarto-dev/quarto-cli#7096] [Q&A]
### Description

I have a template quarto document like this with no code chunks:

```
---
title: "Test"
format: html

knitr:
  opts_chunk:
    results: hold
---
```

Running `quarto render index.qmd` returns this error:

```
▸ quarto render index.qmd

ERROR: Error executing 'Rscript': Not a directory (os error 20)

Unable to locate an installed version of R.
Install R from https://cloud.r-project.org/
```

Removing the knitr argument makes it work. There is no R installation. I understand that knitr is an rmarkdown argument, but it is included as a template option in case an R chunk is included. In this document, there are no R code chunks to actually use that argument. Is this the expected behaviour?

```
▸ quarto --version
1.3.433
```

**Comment by cscheid** [comment]
I believe that if we see options in `knitr:`, we set the engine to `knitr` by default. This is a bit confusing, but it's there (I believe) to reduce the differences between markdown and quarto.

I do agree that it has unfortunate consequences. If I understand the situation correctly, the presence of a top-level `knitr` key in YAMLM metadata stops our engine inference code from firing. You could change this behavior with `engine: markdown`, but then you will not get knitr to run unless there's an explicit `engine: knitr` in the document, which is also unlikely to be what you want.

I can't think of a good solution here, sadly.

**Reply by cderv** [reply]
> if we see options in knitr:, we set the engine to knitr by default (...) but it's there (I believe) to reduce the differences between markdown and quarto.

I wondered the same. I think we set engine when we see `knitr:` because if a document provides such options, then it is meant to be rendered with R engine. 

>  You could change this behavior with engine: markdown

We resolve `knitr: ` field before looking at `execute` engine in `markdownExecutionEngine()` so this won't work. 

Our detection is: 

* look for known engine name in YAML top field
* otherwise, look if `engine:` field is set 
* otherwise, guess from document content with cells language. 
* If no engine found for detected language, defaults to markdown

I don't know why we do the first exactly 🤔 it would probably have been fine without, but now it would break changing this I guess
